satellite/metainfo/metainfo.go

```diff
@@ -506,3 +509,34 @@ func CreatePath(ctx context.Context, projectID uuid.UUID, segmentIndex int64, bu
 	}
 	return storj.JoinPaths(entries...), nil
 }
+
+// checks if bucket has any pointers(entries)
+func (endpoint *Endpoint) checkBucketPointers(ctx context.Context, req *pb.SegmentInfoRequest) (*pb.SegmentInfoResponse, error) {
+	// not_found error indicates connectory id set but no entry exists
+	resp, err := endpoint.SegmentInfo(ctx, req)
+	return resp, err
+}
+
+// ValueAttributeInfo commits segment metadata
+func (endpoint *Endpoint) ValueAttributeInfo(ctx context.Context, req *pb.SegmentInfoRequest) (*pb.SegmentInfoResponse, error) {
+	resp, err := endpoint.checkBucketPointers(ctx, req)
+	if err != nil {
+		switch status.Code(err) {
+		case codes.NotFound:
+			endpoint.log.Sugar().Info("Value attribution entry made\n")
+
+			_, err := endpoint.partnerinfo.Create(ctx, req.GetConnectorKeyInfo())
+			if err != nil {
+				endpoint.log.Sugar().Errorf("Value attribution entry not made: \tpath=", string(req.Path), "\tbucket=", string(req.Bucket), "\terror=", err, "\n")
+				return &pb.SegmentInfoResponse{}, err
+			}
+			return &pb.SegmentInfoResponse{}, nil
+		default:
+			// no entry made into value attribution table
+			endpoint.log.Sugar().Info("Value attribution entry not made: \tpath=", string(req.Path), "\tbucket=", string(req.Bucket), "\terror=", err, "\n")
+			return &pb.SegmentInfoResponse{}, err
+		}
+	}
+
+	return resp, errs.New("BucketID=%v already exists", string(req.GetConnectorKeyInfo().BucketId))
```

Comment:
Shouldn't we support case were bucket exists and its empty?

Reply:
To commit a segment, we need to have a bucket created first. my understanding is calling ValueAttributeInfo() in the bucket creation eliminates lots of redundant checks at a later time of uploads and/or downloads. What do you think?

Reply:
With this part I'm asking if we should have ability to assign value attribution to existing but empty bucket and reject bucket that exists but is not empty. With this code we are able to only set value attribution to non existing bucket.